File: concrete/src/main/java/pcb/uwu/unit/derived/electromagnetism/CoulombUnit.java
package pcb.uwu.unit.derived.electromagnetism;

import pcb.uwu.units.derived.electromagnetism.ElectricChargeUnit;

import static pcb.uwu.unit.base.AmpereUnit.AMPERE;
import static pcb.uwu.unit.base.SecondUnit.SECOND;

public class CoulombUnit extends ElectricChargeUnit {

	public static final CoulombUnit COULOMB = new CoulombUnit();

	public CoulombUnit() {
		super(AMPERE, SECOND);
	}

	@Override
	public String getSymbol() {
		return "C";
	}

	@Override
	public String getSingularName() {
		return "coulomb";
	}

	@Override
	public String getPluralName() {
		return "coulombs";
	}
}

File: implementation/src/main/java/pcb/uwu/unit/derived/electromagnetism/CoulombUnit.kt
package pcb.uwu.unit.derived.electromagnetism

import pcb.uwu.unit.base.AmpereUnit
import pcb.uwu.unit.base.SecondUnit

object CoulombUnit
    : ElectricChargeUnit(electricCurrentUnit = AmpereUnit,
                         timeUnit = SecondUnit)
{
    override val symbol = "C"
    override val singularName = "coulomb"
    override val pluralName = "coulombs"
}
